src/write_influxdb_udp.c

```diff
@@ -364,137 +372,120 @@ static int wifxudp_escape_string(char *buffer, size_t buffer_size,
+    if (status >= (buffer_len - offset))                                       \
+      return offset;                                                           \
   } while (0)
 
 #define BUFFER_ADD(...)                                                        \
   do {                                                                         \
-    status = snprintf(buffer + offset, buffer_len - offset, __VA_ARGS__);      \
-    if ((status < 0) || (status >= (buffer_len - offset))) {                   \
-      sfree(rates);                                                            \
-      return -1;                                                               \
-    }                                                                          \
+    int status = snprintf(buffer + offset, buffer_len - offset, __VA_ARGS__);  \
+    if (status < 0)                                                            \
+      return status;                                                           \
     offset += status;                                                          \
+    if (status >= (buffer_len - offset))                                       \
+      return offset;                                                           \
   } while (0)
 
-  BUFFER_ADD_ESCAPE(vl->plugin);
-  BUFFER_ADD(",host=");
-  BUFFER_ADD_ESCAPE(vl->host);
-  if (strcmp(vl->plugin_instance, "") != 0) {
-    BUFFER_ADD(",instance=");
-    BUFFER_ADD_ESCAPE(vl->plugin_instance);
-  }
-  if (strcmp(vl->type, "") != 0) {
-    BUFFER_ADD(",type=");
-    BUFFER_ADD_ESCAPE(vl->type);
-  }
-  if (strcmp(vl->type_instance, "") != 0) {
-    BUFFER_ADD(",type_instance=");
-    BUFFER_ADD_ESCAPE(vl->type_instance);
+  have_values = false;
+  BUFFER_ADD_ESCAPE(metric.family->name);
+  for (size_t j = 0; j < metric.label.num; j++) {
+    label_pair_t label = metric.label.ptr[j];
+    if (strcmp(label.name, "") != 0 && strcmp(label.value, "") != 0) {
+      BUFFER_ADD(",");
+      BUFFER_ADD_ESCAPE(label.name);
+      BUFFER_ADD("=");
+      BUFFER_ADD_ESCAPE(label.value);
+    }
   }
-
   BUFFER_ADD(" ");
-  for (size_t i = 0; i < ds->ds_num; i++) {
-    if ((ds->ds[i].type != DS_TYPE_COUNTER) &&
-        (ds->ds[i].type != DS_TYPE_GAUGE) &&
-        (ds->ds[i].type != DS_TYPE_DERIVE)) {
-      sfree(rates);
-      return -1;
-    }
 
-    if (ds->ds[i].type == DS_TYPE_GAUGE) {
-      if (isnan(vl->values[i].gauge))
-        continue;
-      if (have_values)
-        BUFFER_ADD(",");
-      BUFFER_ADD("%s=%lf", ds->ds[i].name, vl->values[i].gauge);
+  switch (metric.family->type) {
+  case METRIC_TYPE_GAUGE:
+  case METRIC_TYPE_UNTYPED:
+    if (!isnan(metric.value.gauge)) {
+      BUFFER_ADD("value=" GAUGE_FORMAT, metric.value.gauge);
       have_values = true;
-    } else if (wifxudp_config_store_rates) {
-      if (rates == NULL)
-        rates = uc_get_rate_vl(ds, vl);
-      if (rates == NULL) {
+    }
+    break;
+  default:
```

Comment:
I recommend to use `case METRIC_TYPE_COUNTER` specifically. We're doing to merge the distribution metrics soon, and that would make this switch-case easier to discover, and the code can fail gracefully if it gets called with a distribution metric that it does not yet support.

Reply:
Ok.